### PR TITLE
Clarify RBAC in TargetAllocator readme. Issue ref: #2734

### PR DESCRIPTION
**Pull request by avillela**
**Description:** 

Updated TargetAllocator README to clarify RBAC.

**Link to tracking Issue(s):** 

- Resolves: #2734 

**Testing:**

n/a - no Operator code was modified

**Documentation:**

Updated TargetAllocator README

### CODE REVIEW

File: cmd/otel-allocator/README.md
Comment (nicolastakashi) [inline cmd/otel-allocator/README.md:57]: nit: I think we could link to the prometheus operator repo itself, Since it's the source of truth to the other repos.

- https://github.com/prometheus-operator/prometheus-operator/blob/main/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml

- https://github.com/prometheus-operator/prometheus-operator/blob/main/example/prometheus-operator-crd/monitoring.coreos.com_podmonitors.yaml

Comment (avillela) [reply cmd/otel-allocator/README.md:57]: @nicolastakashi according to the [README of the Prometheus Operator helm chart](https://github.com/prometheus-operator/prometheus-operator/tree/main/helm):

> The helm charts originally developed as part of this repository have been moved to [prometheus-community/kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack).
>
> The multiple charts built in this repository have been combined into a single chart that installs prometheus operator, prometheus, alertmanager, grafana as well as the multitude of exporters necessary to monitor a cluster.

Since that's the case, I would suggest keeping the URLs as-is

**Review by matej-g — APPROVED** [review]
Nice work, thanks for the enhancement 🙇

File: cmd/otel-allocator/README.md
Comment (swiatekm-sumo) [inline cmd/otel-allocator/README.md:150]: I think you have the roles mixed up here. The role below is specifically for Prometheus CRs.

File: .chloggen/avillela-2734.yaml
Comment (swiatekm-sumo) [inline .chloggen/avillela-2734.yaml:1]: We don't normally add changelog entries for documentation updates. Do you think it's worthwhile to have this one?

Comment (pavolloffay) [reply .chloggen/avillela-2734.yaml:1]: +

**Review by pavolloffay — APPROVED** [review]
LGTM,

I would recommend removing the changelog entry before merging

File: .chloggen/avillela-2734.yaml
Comment (avillela) [reply .chloggen/avillela-2734.yaml:1]: The CI build process erred out when I left it out 🤷‍♀️

File: cmd/otel-allocator/README.md
Comment (avillela) [reply cmd/otel-allocator/README.md:150]: Ooops...copy pasta error. Fixed now!

**Comment by pavolloffay** [comment]
> @pavolloffay The CI process erred out when I left it out initially, which is why I put it in. Is there a way to bypass that?

It's expected, maintainers are adding skip changelog label for PRs that don't need a changelog

**Comment by avillela** [comment]
> > @pavolloffay The CI process erred out when I left it out initially, which is why I put it in. Is there a way to bypass that?
> 
> It's expected, maintainers are adding skip changelog label for PRs that don't need a changelog

Ah! TIL 😄

File: .chloggen/avillela-2734.yaml
Comment (avillela) [reply .chloggen/avillela-2734.yaml]: Removed the changelog.

File: cmd/otel-allocator/README.md
Comment (jaronoff97) [inline cmd/otel-allocator/README.md:147]:
```suggestion
If you omit the `ServiceAccount` name, the TargetAllocator creates a `ServiceAccount` for you. The `ServiceAccount`’s default name is a concatenation of the Collector name and the `-targetallocator` suffix. By default, this `ServiceAccount` has no defined policy, so you’ll need to create your own `ClusterRole` and `ClusterRoleBinding` for it, as per below.
```

File: cmd/otel-allocator/README.md
Comment (jaronoff97) [inline cmd/otel-allocator/README.md:132]:
```suggestion
Before the TargetAllocator can start scraping, you need to set up Kubernetes RBAC (role-based access controls) resources. This means that you need to have a `ServiceAccount` and corresponding cluster roles so that the TargetAllocator has access to all of the necessary resources to pull metrics from.
```

**Review by jaronoff97 — COMMENTED** [review]
two small changes, otherwise this looks great!

File: cmd/otel-allocator/README.md
Comment (avillela) [reply cmd/otel-allocator/README.md]: @jaronoff97 I just re-ran my demo, and I noticed that it created a SA with the `-collector` suffix. Is that something different? Per the screen shot below, the `opentelemetry-targetallocator-sa` was created by me, and `otelcol-collector` was automagically created. I don't see any SA with the `-targetallocator` suffix. Just wondering if I'm missing something?

Screenshot 2024-03-11 at 15 33 02

**Pull request merged**
